### PR TITLE
[AC 7393] Update Expert profile pages with accurate labels

### PR DESCRIPTION
**Pull request by iankigen**
## [AC 7393](https://masschallenge.atlassian.net/browse/AC-7393)

**Changes Introduced**:
- Industry  section 
  - merge existing primary and secondary industry sections
  - Displays all industries for the user with the primary industry listed first

- Knowledge areas (previously called ‘Expert mentor in’)

  - Displays all functional expertise listed on the mentor profile

- Program Affiliation (replaces Home Program section)

  - Displays all programs in which the mentor has a confirmed program role grant

  - Home program is displayed at the top of the list with a home icon

This ticket clones [AC 7327](https://masschallenge.atlassian.net/browse/AC-7393): The done definition not implemented in this ticket is provided by AC-7327

**Testing**
- As an admin, Select and save all **Interest categories** this [user](http://localhost:8181/admin/simpleuser/user/20863/change/)
- Masquerade [user](http://localhost:8181/admin/simpleuser/user/20863/masquerade/)
- Visit [user profile page](http://localhost:1234/directory/people/20863)

**Expected behavior**
- **Industry**  section displays all industries for the user with the primary industry listed first
- "Expert mentor in" is renamed to **Knowledge areas**
- **Program Affiliation** Displays all programs in which the mentor has a confirmed program role grant
 and the home program is displayed at the top of the list with a home icon

[Related PR](https://github.com/masschallenge/front-end/pull/222)

### CODE REVIEW

**Comment by jonkiparsky** [comment]
adding @hfreund as a reviewer since he is specified as a required reviewer in the ticket

File: web/impact/impact/tests/test_graphql.py [hunk 1]
```diff
@@ -396,3 +397,29 @@ def test_non_staff_user_can_access_finalist_graphql_view(self):
                     }
                 }
             )
+
+    def test_get_user_confirmed_mentor_program_families(self):
```
Comment (jonkiparsky) [inline web/impact/impact/tests/test_graphql.py:401]: Can we also test the empty case? What if there are no confirmed mentor program families for this user?

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 2]
```diff
@@ -97,6 +101,15 @@ def resolve_current_mentees(self, info, **kwargs):
...
         return _get_mentees(self.user, ENDED_PROGRAM_STATUS)
 
+    def resolve_confirmed_mentor_program_families(self, info, **kwargs):
+        prg = _confirmed_non_future_program_role_grant(self)
+        program_ids = latest_program_id_foreach_program_family()
+        return list(prg.filter(
+            program_role__program__pk__in=program_ids
+        ).values_list(
+            'program_role__program__program_family__name',
+            flat=True).distinct())
```
Comment (jonkiparsky) [inline web/impact/impact/graphql/types/expert_profile_type.py]: Since the Program Affiliation section (per def of done) starts with the Home Program Family, we should probably (a) exclude the home program family from this list and (b) prepend it to the list that we return. Then your front-end code can know that the first item in the list will be their home program family, and decorate it appropriately

**Review by jonkiparsky — CHANGES_REQUESTED** [review]

**Review by hfreund — APPROVED** [review]
Leaving product approval even though there isn't much product related information to review on this PR. This will still require two engineer reviews.

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 3]
```diff
@@ -97,6 +101,15 @@ def resolve_current_mentees(self, info, **kwargs):
     def resolve_previous_mentees(self, info, **kwargs):
         return _get_mentees(self.user, ENDED_PROGRAM_STATUS)
 
+    def resolve_confirmed_mentor_program_families(self, info, **kwargs):
+        prg = _confirmed_non_future_program_role_grant(self)
+        program_ids = latest_program_id_foreach_program_family()
+        return list(prg.filter(
```
[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: actually not, removing that

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 4]
```diff
@@ -131,3 +144,15 @@ def _get_mentees(user, program_status):
         status=CONFIRMED_RELATIONSHIP,
         **mentee_filter
     ).order_by('-startup_mentor_tracking__program__start_date')
+
+
+def _confirmed_non_future_program_role_grant(expert_profile):
```
[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: switching to ` _visible_confirmed_mentor_role_grants`

File: web/impact/impact/v1/helpers/profile_helper.py [hunk 5]
```diff
@@ -512,7 +512,7 @@ def is_non_member(self):
     @property
     def confirmed_user_program_families(self):
         prg = _confirmed_non_future_program_role_grant(self.subject)
-        program_ids = _latest_program_id_foreach_program_family()
+        program_ids = latest_program_id_foreach_program_family()
```
[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/v1/helpers/profile_helper.py]: yeah, on it

File: web/impact/impact/v1/helpers/profile_helper.py [hunk 6]
```diff
@@ -549,7 +549,7 @@ def _latest_confirmed_non_future_program_role_grant(obj):
     return prg.order_by('-created_at').first()
 
 
-def _latest_program_id_foreach_program_family():
+def latest_program_id_foreach_program_family():
```
[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/v1/helpers/profile_helper.py]: This function has the same functionality as [def _program_family__program_subquery](https://github.com/masschallenge/accelerate/blob/98ce27c34be6dd697dab5320948fb9f1b1baa8b6/mcproject/mc/models/expert_profile.py#L252) from accelerate, 
I copied the same functionality over to impact for consistency (ends up populating `confirmed_mentor_program_families` of algolia index) as there was no clear way of importing it

File: web/impact/impact/graphql/types/expert_profile_type.py [hunk 7]
```diff
@@ -97,6 +101,15 @@ def resolve_current_mentees(self, info, **kwargs):
     def resolve_previous_mentees(self, info, **kwargs):
         return _get_mentees(self.user, ENDED_PROGRAM_STATUS)
 
+    def resolve_confirmed_mentor_program_families(self, info, **kwargs):
+        prg = _confirmed_non_future_program_role_grant(self)
```
[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: I agree, however in this case self is `ExpertProfile` instance and `confirmed_non_future_program_role_grant` is not one of its methods.

[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py]: Hm, that sounds really wrong to me. Someone must be doing some bad juju if that's true. Have you tried it?

[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: yeah, https://docs.graphene-python.org/en/stable/types/objecttypes/#naming-convention
It's because we are subclassing ObjectType

[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py]: Oh, that's horrible. Hulk smash.

[the remark this answers is not in the document]
Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py]: Can we just rename that `self` to something else (as suggested in those docs), since it's not actually a reference to `self`?

[the remark this answers is not in the document]
Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: yeah, renaming to `obj`

**Review by freshprincekla — APPROVED** [review]
QA passes.
- I was able to visit the Expert Profiles and see the accurate labels
- The industry Section displays all the user's industries with primary industry displayed first
- `Expert mentor in` was renamed to `Knowledge areas`
- `Program Affiliation` replaced `Home Program`, with the home program first with a home icon.

File: web/impact/impact/graphql/types/expert_profile_type.py (hunk 2 above)
Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py]: > Since the Program Affiliation section (per def of done) starts with the Home Program Family, we should probably (a) exclude the home program family from this list and (b) prepend it to the list that we return. Then your front-end code can know that the first item in the list will be their home program family, and decorate it appropriately

@iankigen I notice that you gave this comment a positive reaction, but have not acted on it. Are you intending to rearrange the code as suggested? Obviously, I would prefer to see this change, because it will make the code simpler all around, but I will understand if you want to push back on it.

Comment (iankigen) [reply web/impact/impact/graphql/types/expert_profile_type.py]: @jonkiparsky I have [excluded](https://github.com/masschallenge/impact-api/blob/7be69ae8f2cb8d39e8eabe32bd0cd696604e0559/web/impact/impact/graphql/types/expert_profile_type.py#L107-L113) the home program family from the list. On the front end, I'm prepending the `home program`  to `confirmed mentor program families` list

Comment (jonkiparsky) [reply web/impact/impact/graphql/types/expert_profile_type.py]: Gotcha, missed that (I blame github :) )

**Review by jonkiparsky — APPROVED** [review]
Code much improved, thanks. QA looks fine as well.
